torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -354,6 +354,178 @@ def layer_norm_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+        # pointwise result of x_hat and weight/bias.
+        if output_mask[0]:
+            # TODO: now grad_out spec follows input spec. we may need
+            # to change it to apply a pointwise rule over grad_out,
+            # input, and weight.
+            grad_out_target_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(input_src_spec.placements, axis),
+                tensor_meta=input_src_spec.tensor_meta,
+            )
+            op_args_target_specs.append(grad_out_target_spec)
+            redistribute_costs.append(
+                generate_redistribute_costs(grad_out_strategy, grad_out_target_spec)
+            )
+            output_specs_list.append(grad_out_target_spec)
+            out_tuple_strategy.strategies.append(
+                # these 3 lists will be filled in later
+                PlacementStrategy(
+                    output_spec=output_specs_list,
+                    input_specs=op_args_target_specs,
+                    redistribute_cost=redistribute_costs,
+                )
+            )
+        else:
+            output_specs_list.append(None)
+
+        # arg: input
+        input_target_spec = DTensorSpec(
+            mesh=mesh,
+            placements=_replicate_dims_start_at(input_src_spec.placements, axis),
+            tensor_meta=input_src_spec.tensor_meta,
+        )
+        op_args_target_specs.append(input_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(input_strategy, input_target_spec)
+        )
+
+        # arg: mean, rstd
+        mean_target_spec = mean_strategy.strategies[idx].output_spec
+        assert isinstance(mean_target_spec, DTensorSpec)
+        op_args_target_specs.append(mean_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(mean_strategy, mean_target_spec)
+        )
+        rstd_target_spec = rstd_strategy.strategies[idx].output_spec
+        assert isinstance(rstd_target_spec, DTensorSpec)
+        op_args_target_specs.append(rstd_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(rstd_strategy, rstd_target_spec)
+        )
+
+        # arg: weight
+        # d_weight = sum(grad_out * (input - mean) / rstd, outer_dim, keepdim=False)
+        if output_mask[1]:
+            assert isinstance(weight_strategy, OpStrategy)
+            weight_src_spec = weight_strategy.strategies[idx].output_spec
+            assert isinstance(weight_src_spec, DTensorSpec)
+            weight_target_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(weight_src_spec.placements),
+                tensor_meta=weight_src_spec.tensor_meta,
+            )
+            op_args_target_specs.append(weight_target_spec)
+            redistribute_costs.append(
+                generate_redistribute_costs(weight_strategy, weight_target_spec)
```

Comment:
same, weight spec redistribute cost seems 0 too

Reply:
Because it's always "Replicate"?

Reply:
because the weight_target_spec come from weight_strategy (one of the weight strategy), which means they should be idential

Reply:
oh that's right. I keep confusing `weight` and `d_weight`...